AEPCore/Sources/eventhub/SDKInstanceIdentifier.swift

```diff
@@ -0,0 +1,77 @@
+//
+/*
+ Copyright 2024 Adobe. All rights reserved.
+ This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License. You may obtain a copy
+ of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software distributed under
+ the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ OF ANY KIND, either express or implied. See the License for the specific language
+ governing permissions and limitations under the License.
+ */
+
+import Foundation
+
+/// An identifier for an SDK instance.
+public enum SDKInstanceIdentifier: Hashable {
+    case `default`
+    case id(String)
+    
+    
+    /// Creates an instance of `SDKInstanceIdentifier`. 
+    /// If the given `id` is either too long or contains invalid characters then `nil` is returned.
+    /// - Parameter id: the identifier string for an SDK instance.
+    init?(id: String) {
+        if id == "default-instance" {
+            self = .default
+        } else if !SDKInstanceIdentifier.isValidFilename(id) {
```

Comment:
nit: Can you invert the condition?
} else if SDKInstanceIdentifier.isValidFilename(id) {

} else  {
   return nil
}